src/Workspaces/Core/Portable/LanguageServices/TypeInferenceService/AbstractTypeInferenceService.cs

```diff
@@ -74,6 +74,21 @@ private IEnumerable<ITypeSymbol> Filter(IEnumerable<ITypeSymbol> types, bool fil
                             .Distinct()
                             .ToImmutableReadOnlyListOrEmpty();
             }
+
+            protected IEnumerable<ITypeSymbol> ExpandParamsParameter(IParameterSymbol parameterSymbol)
+            {
+                var result = SpecializedCollections.SingletonEnumerable(parameterSymbol.Type);
+                if (parameterSymbol.IsParams)
+                {
+                    var arrayTypeSymbol = parameterSymbol.Type as IArrayTypeSymbol;
+                    if (arrayTypeSymbol != null)
+                    {
+                        result = result.Concat(arrayTypeSymbol.ElementType);
```

Comment:
why not just have a list you can add to?  i'm not a fan of concatting lists to lists.